uniffi_aries_vcx/demo/app/src/main/java/org/hyperledger/ariesvcx/ScanScreen.kt

```diff
@@ -0,0 +1,162 @@
+package org.hyperledger.ariesvcx
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.util.Size
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.Preview
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.camera.view.PreviewView
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.core.content.ContextCompat
+import android.util.Base64
+import androidx.compose.runtime.rememberCoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.json.JSONObject
+
+@Composable
+fun ScanScreen(connection: Connection?, profileHolder: ProfileHolder?) {
+    var scannedQRCodeText by remember {
+        mutableStateOf<String?>(null)
+    }
+    val scope = rememberCoroutineScope()
+
+    val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val cameraProviderFuture = remember {
+        ProcessCameraProvider.getInstance(context)
+    }
+
+    scannedQRCodeText?.let {text ->
+        val encoded = Uri.parse(text)?.getQueryParameter("c_i")
+        val decoded =  String(Base64.decode(encoded, Base64.DEFAULT))
+        val map = jsonStringToMap(decoded)
+
+        AlertDialog(
+            onDismissRequest = { scannedQRCodeText = null },
+            title = { Text("Accept this invitation?") },
+            text = { Text(decoded) },
+            confirmButton = {
+                TextButton(onClick = {
+                    scope.launch(Dispatchers.IO) {
+                        if (profileHolder != null) {
+                            connection?.acceptInvitation(
+                                profile = profileHolder,
+                                invitation = decoded
+                            )
+                            connection?.sendRequest(profileHolder, map["serviceEndpoint"] as String, map["routingKeys"] as List<String>)
```

Comment:
this is not the correct usage of `sendRequest`. `send_request` in aries_vcx takes in the service endpoint and routing keys that you own. Here you are passing in the endpoint and keys that the Inviter owns (bcus you're extracting them from the Inviter's invitation).

As such, it's probably best to just leave the serviceEndpoint and routingKeys as dummy inputs for now: e.g. `https://google.com` and `emptyList()`.

In your next PR we will worry about replacing those dummy inputs with real values (where the real values will involve the simple relay service i've mentioned before)